drivers/ethernet/eth_numaker.c

```diff
@@ -0,0 +1,807 @@
+
+	delay_cnt = 3000000U;
+	ret = WAIT_FOR((mdio_read(gmacdev, eth_phy_addr, MII_BMSR) &
+					(BMSR_ANEGCOMPLETE | BMSR_LSTATUS)) ==
+					(BMSR_ANEGCOMPLETE | BMSR_LSTATUS),
+					delay_cnt, k_msleep(1));
+
+	if (ret == false) {
+		LOG_DBG("AN failed. Set to 100 FULL");
+		synopGMAC_set_full_duplex(gmacdev);
+		synopGMAC_set_mode(NUMAKER_GMAC_INTF, 1); /* Set mode 1: 100Mbps; 2: 10Mbps */
+		return -ENODATA;
+	}
+	reg = mdio_read(gmacdev, eth_phy_addr, MII_LPA);
+	if (reg & ADVERTISE_100FULL) {
+		LOG_DBG("100 full");
+		gmacdev->DuplexMode = FULLDUPLEX;
+		gmacdev->Speed = SPEED100;
+		synopGMAC_set_full_duplex(gmacdev);
+		synopGMAC_set_mode(NUMAKER_GMAC_INTF, 1); /* Set mode 1: 100Mbps; 2: 10Mbps */
+	} else if (reg & ADVERTISE_100HALF) {
+		LOG_DBG("100 half");
+		gmacdev->DuplexMode = HALFDUPLEX;
+		gmacdev->Speed = SPEED100;
+		synopGMAC_set_half_duplex(gmacdev);
+		synopGMAC_set_mode(NUMAKER_GMAC_INTF, 1); /* Set mode 1: 100Mbps; 2: 10Mbps */
+	} else if (reg & ADVERTISE_10FULL) {
+		LOG_DBG("10 full");
+		gmacdev->DuplexMode = FULLDUPLEX;
+		gmacdev->Speed = SPEED10;
+		synopGMAC_set_full_duplex(gmacdev);
+		synopGMAC_set_mode(NUMAKER_GMAC_INTF, 2); /* Set mode 1: 100Mbps; 2: 10Mbps */
+	} else {
+		LOG_DBG("10 half");
+		gmacdev->DuplexMode = HALFDUPLEX;
+		gmacdev->Speed = SPEED10;
+		synopGMAC_set_half_duplex(gmacdev);
+		synopGMAC_set_mode(NUMAKER_GMAC_INTF, 2); /* Set mode 1: 100Mbps; 2: 10Mbps */
+	}
+
+	LOG_INF("PHY ID 1:0x%x", mdio_read(gmacdev, eth_phy_addr, MII_PHYSID1));
+	LOG_INF("PHY ID 2:0x%x", mdio_read(gmacdev, eth_phy_addr, MII_PHYSID2));
+
+	return status;
+}
+
+static void m_numaker_read_mac_addr(char *mac)
+{
+	uint32_t uid1;
+	/* Fetch word 0 of data flash */
+	uint32_t word0 = *(uint32_t *)(NUMAKER_DATA_FLASH + 0x04U);
+	/*
+	 * Fetch word 1 of data flash
+	 * we only want bottom 16 bits of word1 (MAC bits 32-47)
+	 * and bit 9 forced to 1, bit 8 forced to 0
+	 * Locally administered MAC, reduced conflicts
+	 * http://en.wikipedia.org/wiki/MAC_address
+	 */
+	uint32_t word1 = *(uint32_t *)NUMAKER_DATA_FLASH;
+
+	/* Not burn any mac address at the beginning of data flash */
+	if (word0 == NUMAKER_MASK_32) {
+		/* Generate a semi-unique MAC address from the UUID */
+		SYS_UnlockReg();
+		/* Enable FMC ISP function */
+		FMC_Open();
+		uid1 = FMC_ReadUID(1);
+		word1 = (uid1 & 0x003FFFFF) | ((uid1 & 0x030000) << 6) >> 8;
+		word0 = ((FMC_ReadUID(0) >> 4) << 20) | ((uid1 & 0xFF) << 12) |
+			(FMC_ReadUID(2) & 0xFFF);
+		/* Disable FMC ISP function */
+		FMC_Close();
+		/* Lock protected registers */
+		SYS_LockReg();
+	}
+
+	word1 |= 0x00000200;
+	word1 &= 0x0000FEFF;
+
+	mac[0] = (word1 & 0x0000ff00) >> 8;
+	mac[1] = (word1 & 0x000000ff);
+	mac[2] = (word0 & 0xff000000) >> 24;
+	mac[3] = (word0 & 0x00ff0000) >> 16;
+	mac[4] = (word0 & 0x0000ff00) >> 8;
+	mac[5] = (word0 & 0x000000ff);
+
+	LOG_INF("mac address %02x-%02x-%02x-%02x-%02x-%02x", mac[0], mac[1], mac[2], mac[3],
```

Comment:
`%02x:%02x:%02x:%02x:%02x:%02x`? That's how field in mac addrss are normally separated

Reply:
Oh, I see, it's esparted by : in Linux.
However, in Windows, it's separated by "-". Zephyr is Linux style, it's fine to follow it.

![image](https://github.com/zephyrproject-rtos/zephyr/assets/14323432/c815cefb-3c2a-422a-9765-2fdb6b789ed9)

Reply:
ha, funny, sorry haven't touched a Windows machine in a long time :-) I did a git grep for that pattern in the Zephyr code base (`git grep %02x.%02x.%02x.%02x.%02x.%02x`) seems like the rest of the code uses `:` consistently, but that's an interesting observation.